### ISSUE TITLE
Nullness Checker crash: AsSuperVisitor: type is not an erased subtype of supertype 10

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#1419]
In a repo with master and #1387 take:

```java
abstract class Crash13 {
  class Map<K, V> {}
  class EnumMap<K extends Enum<K>, V> extends Map<K, V> {}

  abstract <K extends Enum<K>, V> Map<K, V> foo(Map<K, ? extends V> map);

  <K, V> Map<K, V> bar(Map<? extends K, ? extends V> map) {
    return foo((EnumMap) map);
  }
}
```

Running the Nullness Checker gives:

```
error: AsSuperVisitor: type is not an erased subtype of supertype.
  type: @Initialized @Nullable Object
  superType: @Initialized @NonNull Enum<K extends @Initialized @Nullable Object>
  Compilation unit: Crash13.java
  Exception: java.lang.Throwable; Stack trace: org.checkerframework.framework.source.SourceChecker.errorAbort(SourceChecker.java:727)
  org.checkerframework.javacutil.ErrorReporter.errorAbort(ErrorReporter.java:41)
  org.checkerframework.framework.type.AsSuperVisitor.errorTypeNotErasedSubtypeOfSuperType(AsSuperVisitor.java:155)
  org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:340)
  org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:29)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:308)
  org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:59)
  org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:86)
  org.checkerframework.framework.type.AsSuperVisitor.asSuper(AsSuperVisitor.java:68)
  org.checkerframework.framework.util.AnnotatedTypes.asSuper(AnnotatedTypes.java:131)
  org.checkerframework.framework.type.DefaultTypeHierarchy.castedAsSuper(DefaultTypeHierarchy.java:1112)
  org.checkerframework.framework.type.DefaultTypeHierarchy.visitDeclared_Declared(DefaultTypeHierarchy.java:485)
  org.checkerframework.framework.type.DefaultTypeHierarchy.visitDeclared_Declared(DefaultTypeHierarchy.java:55)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:308)
  org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:238)
  org.checkerframework.framework.type.DefaultTypeHierarchy.checkAndSubtype(DefaultTypeHierarchy.java:330)
  org.checkerframework.framework.type.DefaultTypeHierarchy.visitTypevarSubtype(DefaultTypeHierarchy.java:1030)
  org.checkerframework.framework.type.DefaultTypeHierarchy.visitTypevar_Declared(DefaultTypeHierarchy.java:858)
  org.checkerframework.framework.type.DefaultTypeHierarchy.visitTypevar_Declared(DefaultTypeHierarchy.java:55)
  org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:632)
  org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:238)
  org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:173)
  org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:151)
  org.checkerframework.common.basetype.BaseTypeVisitor.commonAssignmentCheck(BaseTypeVisitor.java:1814)
  org.checkerframework.checker.nullness.NullnessVisitor.commonAssignmentCheck(NullnessVisitor.java:221)
  org.checkerframework.common.basetype.BaseTypeVisitor.checkTypeArguments(BaseTypeVisitor.java:1991)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:935)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:160)
  com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1477)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
```

**Comment by smillst** [comment]
Fixed in #1419.

**Comment by smillst** [comment]
Fixed via a03f6ba.
